crates/vm/src/system/memory/controller/mod.rs

```diff
@@ -379,64 +339,85 @@ impl<F: PrimeField32> MemoryController<F> {
         (record_id, data)
     }
 
-    pub fn read<const N: usize>(&mut self, address_space: F, pointer: F) -> (RecordId, [F; N]) {
+    // TEMP[jpw]: Function is safe temporarily for refactoring
+    /// # Safety
+    /// The type `T` must be stack-allocated `repr(C)` or `repr(transparent)`, and it must be the
+    /// exact type used to represent a single memory cell in address space `address_space`. For
+    /// standard usage, `T` is either `u8` or `F` where `F` is the base field of the ZK backend.
+    pub fn read<T: Copy, const N: usize>(
```

Comment:
@shuklaayush @Golovanov399 is it now safe to delete these read functions from `MemoryController`?